app/_components/_sections/SavedWorkoutsList.tsx

```diff
@@ -0,0 +1,196 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { FaArrowLeftLong } from "react-icons/fa6";
+import { CiBookmarkMinus } from "react-icons/ci";
+import { CgNotes } from "react-icons/cg";
+import { HiMiniSquares2X2 } from "react-icons/hi2";
+import { FaSpinner } from "react-icons/fa";
+import DeleteWorkoutModal from "../modals/deleteWorkoutModal";
+import { useTranslation } from "react-i18next";
+import { useDeleteWorkout } from "@/app/_services/userService";
+
+interface Workout {
+  _id: string;
+  image_url: string;
+  name: string;
+  level: string;
+  duration?: number;
+  workout_schedule: any[];
+}
+
+interface SavedWorkoutsModalProps {
+  workouts: Workout[];
+  onClose: () => void;
+  refetchSavedWorkouts: () => void;
+}
+
+const calculateTotalDuration = (workout: Workout) => {
+  let totalDurationSeconds = 0;
+
+  workout.workout_schedule?.forEach((schedule) => {
+    schedule.exercises?.forEach((exercise) => {
+      const sets = exercise.sets || 0;
+      const restSeconds = exercise.rest_seconds || 0;
+      const setDurationSeconds = 120;
+
+      const exerciseDurationSeconds = sets * setDurationSeconds;
+      const totalRestBetweenSets = (sets - 1) * restSeconds;
+
+      totalDurationSeconds += exerciseDurationSeconds + totalRestBetweenSets;
+    });
+  });
+
+  return Math.floor(totalDurationSeconds / 60);
+};
+
+const SavedWorkoutsModal: React.FC<SavedWorkoutsModalProps> = ({ workouts, onClose, refetchSavedWorkouts }) => {
+  const { t } = useTranslation("global");
+  const { mutate: deleteWorkout } = useDeleteWorkout();
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedWorkout, setSelectedWorkout] = useState<Workout | null>(null);
+  const [clickedStates, setClickedStates] = useState<{ [key: string]: boolean }>({});
```

Comment:
Issue: This state duplicates information that can be derived directly from the workoutList.
Solution: Instead of keeping separate states, infer the clicked state from the workout list.

`const isBookmarked = workoutList.some((w) => w._id === workout._id);
`